lib/ssl/test/ssl_bench_SUITE.erl

```diff
@@ -89,73 +72,112 @@ end_per_testcase(_Func, _Conf) ->
     ok.
 
-payload_simple(Config) ->
+setup_concurrent_noreuse(Config) ->
     Server = proplists:get_value(server_node, Config),
     Server =/= undefined orelse error(no_server),
-    {ok, Result} = do_test(ssl, payload, ?COUNT*300, 10, Server),
+    Cfg = [{version, 'tlsv1.2'}, no_reuse],
+    {ok, Result} = do_test(ssl, {setup_connection,Cfg}, ?COUNT, 100, Server),
     ct_event:notify(#event{name = benchmark_data,
 			   data=[{value, Result},
-				 {suite, "ssl"}, {name, "Payload simple"}]}),
+				 {suite, "ssl"}, {name, "Conc setup 1.2 no session"}]}),
     ok.
 
-basic_pem_cache(_Config) ->
-    do_test(ssl, pem_cache, 10, 5, node()).
-
-use_pem_cache(_Config) ->
-    {ok, Result} = do_test(ssl, pem_cache, 100, 500, node()),
+setup_concurrent_13(Config) ->
+    Server = proplists:get_value(server_node, Config),
+    Server =/= undefined orelse error(no_server),
+    Cfg = [{version, 'tlsv1.3'}],
+    {ok, Result} = do_test(ssl, {setup_connection,Cfg}, ?COUNT, 100, Server),
     ct_event:notify(#event{name = benchmark_data,
-                           data=[{value, Result},
-                                 {suite, "ssl"}, {name, "Use PEM cache"}]}).
+			   data=[{value, Result},
+				 {suite, "ssl"}, {name, "Conc setup 1.3"}]}),
+    ok.
 
-bypass_pem_cache(_Config) ->
-    {ok, Result} = do_test(ssl, pem_cache, 100, 500, node()),
+payload(Config) ->
+    Server = proplists:get_value(server_node, Config),
+    Server =/= undefined orelse error(no_server),
+    Cfg = [{version, 'tlsv1.2'}],
+    {ok, Result} = do_test(ssl, {payload, Cfg}, ?COUNT*300, 10, Server),
     ct_event:notify(#event{name = benchmark_data,
-                           data=[{value, Result},
-                                 {suite, "ssl"}, {name, "Bypass PEM cache"}]}).
+			   data=[{value, Result},
+				 {suite, "ssl"}, {name, "Payload simple"}]}),
+    ok.
 
+payload_13(Config) ->
+    Server = proplists:get_value(server_node, Config),
+    Server =/= undefined orelse error(no_server),
+    Cfg = [{version, 'tlsv1.3'}],
+    {ok, Result} = do_test(ssl, {payload, Cfg}, ?COUNT*300, 10, Server),
+    ct_event:notify(#event{name = benchmark_data,
+			   data=[{value, Result},
+				 {suite, "ssl"}, {name, "Payload 1.3"}]}),
+    ok.
 
 ssl() ->
     test(ssl, ?COUNT).
 
 test(Type, Count) ->
     Server = ssl_bench_test_lib:setup(perf_server),
-    (do_test(Type, setup_connection, Count * 20, 1, Server)),
-    (do_test(Type, setup_connection, Count, 100, Server)),
-    (do_test(Type, payload, Count*300, 10, Server)),
+    (do_test(Type, {setup_connection, [{version, 'tlsv1.2'}]}, Count * 20, 1, Server)),
+    (do_test(Type, {setup_connection, [{version, 'tlsv1.3'}]}, Count * 20, 1, Server)),
+    (do_test(Type, {setup_connection, [{version, 'tlsv1.2'}, no_reuse]}, Count * 20, 1, Server)),
+    (do_test(Type, {setup_connection, [{version, 'tlsv1.2'}]}, Count, 100, Server)),
+    (do_test(Type, {payload, [{version, 'tlsv1.2'}]}, Count*300, 10, Server)),
```

Comment:
I think we should add a test for TLS-1.3 with session tickets, but it can be the next step.

Reply:
I'll leave that to Peter :-)